mlir/lib/Dialect/LLVMIR/IR/LLVMLinkerInterface.cpp

```diff
@@ -143,6 +141,14 @@ StringRef symbol(Operation *op) {
   llvm_unreachable("unexpected operation");
 }
 
+StringAttr symbolAttr(Operation *op) {
+  if (auto gv = dyn_cast<LLVM::GlobalOp>(op))
+    return gv.getSymNameAttr();
+  if (auto fn = dyn_cast<LLVM::LLVMFuncOp>(op))
+    return fn.getSymNameAttr();
```

Comment:
Why this instead of using `SymbolOpInterface::getNameAttr()`? Are there operations with symbols that are not going to be linked?